.github/workflows/integration.yml

```diff
@@ -0,0 +1,321 @@
+          cache: 'yarn'
+
+      - name: Installing Dependencies
+        run: yarn install --ignore-engines --frozen-lockfile
+
+      - name: Compiling Code
+        run: yarn build
+
+      - name: Mocha Tests Azure
+        run: yarn cover-azure
+
+      - name: Performance Tests
+        run: sh test.sh
+
+      - name: Codecov
+        run: yarn codecov
+
+  test-google:
+    name: Test Google
+    needs: test-local
+    runs-on: ubuntu-latest
+    env:
+      WINSTON_CONSOLE_LEVEL: none
+      EXPRESS_PORT: 1337
+      STORAGE_REPO: google
+      FS_GOOGLE_CLOUD_BUCKET: ${{ secrets.FS_GOOGLE_CLOUD_BUCKET }}
+      FS_GOOGLE_CLOUD_PROJECT_ID: ${{ secrets.FS_GOOGLE_CLOUD_PROJECT_ID }}
+      GCLOUD_SERVICE_KEY: ${{ secrets.GCLOUD_SERVICE_KEY }}
+
+    services:
+      mongodb:
+        image: mongo:3.7.9
+        env:
+          MONGO_INITDB_DATABASE: test
+          ALLOW_EMPTY_PASSWORD: yes
+        ports:
+          - 27017:27017
+        options: --health-cmd="mongo" --health-interval=10s --health-timeout=5s --health-retries=5
+
+      redis:
+        image: redis:3.2.12
+        env:
+          REDIS_DATABASE: test
+          ALLOW_EMPTY_PASSWORD: yes
+        ports:
+          - 6379:6379
+        options: --health-cmd="redis-cli ping"
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+          cache: 'yarn'
+
+      - name: Installing Dependencies
+        run: yarn install --ignore-engines --frozen-lockfile
+
+      - name: Compiling Code
+        run: yarn build
+
+      - name: Setup Google Key File
+        run: echo ${GCLOUD_SERVICE_KEY} > ./google.keyfile.json
+
+      - name: Mocha Tests Google
+        run: yarn cover-google
+
+      - name: Performance Tests
+        run: sh test.sh
+
+      - name: Codecov
+        run: yarn codecov
+
+  release:
+    name: Release
+    if: github.ref == 'refs/heads/master'
+    needs: [test-local, test-s3]
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+          cache: 'yarn'
+
+      - name: Installing Dependencies
+        run: yarn install --ignore-engines --frozen-lockfile
+
+      - name: Run Semantic Release
+        run: npm run semantic-release
+        env:
+          GITHUB_TOKEN: ${{ secrets.SEMANTIC_RELEASE_GITHUB_TOKEN }}
+
+  docker-build:
+    name: Docker Build
+    if: (!contains(github.ref_name, 'renovate') && !contains(github.ref_name, 'dependabot'))
+    needs: [test-local, test-s3]
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Installing Dependencies
+        run: yarn install --ignore-engines --frozen-lockfile
+
+      - name: Compiling Code
+        run: yarn build
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```

Comment:
Ya, wasn't sure what the changes were from v1 to v2, and no time now to investigate, we know v1 does the job here because we use it elsewhere. Something we can get Niall to loop back around